### PR TITLE
set tag for zero inl case

### PR DESCRIPTION
**Pull request by kojo1**

### CODE REVIEW

**Review by julek-wolfssl — CHANGES_REQUESTED** [review]
It looks good. I didn't think of the zero length encryption case. My only comment would be to add a known answer test in api.c: please run a `EVP_EncryptInit_ex` -> `EVP_EncryptFinal_ex` -> `EVP_CIPHER_CTX_ctrl(EVP_CTRL_AEAD_GET_TAG)` cycle with OpenSSL and make sure that we arrive at the same result. The OpenSSL result should be saved and compared to our result (for the same input parameters) in `./tests/api.c`.

**Comment by julek-wolfssl** [comment]
Known answer tests for both encryption and decryption please. Thanks.

**Review by julek-wolfssl — APPROVED** [review]

File: tests/api.c [hunk 1]
```diff
@@ -36487,6 +36487,55 @@ static void test_wolfssl_EVP_aes_gcm_AAD_2_parts(void)
 #endif
 }
 
+#if defined(OPENSSL_EXTRA) && !defined(NO_AES) && defined(HAVE_AESGCM) && \
+    !defined(HAVE_SELFTEST) && !defined(HAVE_FIPS)
+static void test_wolfssl_EVP_aes_gcm_zeroLen()
+{       
+    /* Zero length plain text */
+
+    byte key[] = {0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00};  /* align */
```
Comment (cconlon) [inline tests/api.c:36496]: Can we re-format this array to maintain our coding standard 80-character line max?

File: tests/api.c [hunk 2]
```diff
@@ -36487,6 +36487,55 @@ static void test_wolfssl_EVP_aes_gcm_AAD_2_parts(void)
...
+    byte iv[]  = {0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00};  /* align */
```
Comment (cconlon) [inline tests/api.c:36497]: Same as above, please check 80-character line limit

File: tests/api.c [hunk 3]
```diff
@@ -36487,6 +36487,55 @@ static void test_wolfssl_EVP_aes_gcm_AAD_2_parts(void)
...
+    byte plaintxt[0];
+    int ivSz  = 12;
+    int plaintxtSz = 0;
+    unsigned char tag[16];
+    unsigned char tag_kat[] = {0x53,0x0f,0x8a,0xfb,0xc7,0x45,0x36,0xb9,0xa9,0x63,0xb4,0xf1,0xc4,0xcb,0x73,0x8b};
```
Comment (cconlon) [inline tests/api.c:36502]: Same as above, please check 80-character line limit

**Review by cconlon — CHANGES_REQUESTED** [review]

File: tests/api.c (hunk 3 above)
Comment (wolfssl-jp) [reply tests/api.c]: folded it

File: tests/api.c (hunk 2 above)
Comment (wolfssl-jp) [reply tests/api.c]: Folded it

File: tests/api.c (hunk 1 above)
Comment (wolfssl-jp) [reply tests/api.c]: Fixed

**Review by cconlon — APPROVED** [review]

**Pull request merged**